### PR TITLE
Bring back the broken tests.

### PR DESCRIPTION
**Pull request by RoySegall**
#402

Let's see what's going on here.

### CODE REVIEW

**Comment by e0ipso** [comment]
It seems to be failing. We should keep the dependency of RESTful 7.x-1.x with Entity Validator 7.x-1.x

**Comment by RoySegall** [comment]
I created a new isssue - https://github.com/Gizra/entity_validator/issues/39. I can try to use the 1.1 branch and see how the tests affected. If so - i'll remove the FieldsInfo idea and get back to normal array.

**Comment by amitaibu** [comment]
@RoySegall note that currently on `7.x-1.x` the tests are commented out, so you will need to re-add them in the `.info` file

**Comment by RoySegall** [comment]
@amitaibu I saw, but i branched out before you comment them out.

**Comment by amitaibu** [comment]
Ok, anyway I think it's best to create the PR on top of the updated 1.x

**Comment by RoySegall** [comment]
There is something wrong with the node access. I'll check this tomorrow.

**Comment by e0ipso** [comment]
Thanks Roy!

**Comment by RoySegall** [comment]
Well i think i found something that might be relate to this one. I'm writing tests for OpenScholar restful part and i update the entity module to the dev version in order to avoid the errors of undefined nid. As john, he is a group manager - he have permission to create publication node type, restful return access denied when trying to post. I went to the publication form and i managed to create publication node type.

This dissonance throw me back to my previous comment when i mentioned the node access issue.

**Comment by RoySegall** [comment]
@amitaibu @mateu-aguilo-bosch Fixed.

**Comment by e0ipso** [comment]
What was the issue?

**Comment by RoySegall** [comment]
I set the public fields info wrong and another problem that we didn't find out before. There is another PR for the entity validator that wait to be merged in so don't merge this fix.

**Comment by RoySegall** [comment]
@mateu-aguilo-bosch @amitaibu I undo the node byapss and i get in the tests `RestfulBadRequestException: Property label cannot be set. in RestfulEntityBase->setPropertyValues() (line 668 of /var/www/html/drupal/sites/all/modules/restful/plugins/restful/RestfulEntityBase.php).`

This is still need some work.

**Comment by RoySegall** [comment]
I found the problem. The entity validator was not the only one that did all this fiasco. The thing is: the patch for removing the notices always passes `create` to the node access as the operation. When the node is new and don't own any node id `node_access()` crashes.

**Comment by RoySegall** [comment]
@amitaibu Fixed and ready for merge.

File: .travis.yml [hunk 1]
```diff
@@ -30,15 +30,16 @@ before_script:
...
   # Download entity validator.
   - drush dl entity_validator --dev --yes
 
   # Patch Entity API.
-  - curl -O https://www.drupal.org/files/issues/2086225-entity-access-check-node-create-3.patch
-  - patch -p1 /home/travis/build/restful/drupal/sites/all/modules/entity/modules/callbacks.inc  < 2086225-entity-access-check-node-create-3.patch
-
-  - curl -O https://www.drupal.org/files/issues/2264079-entity-wrapper-access-single-entity-reference-2.patch
-  - patch -p1 /home/travis/build/restful/drupal/sites/all/modules/entity/includes/entity.wrapper.inc < 2264079-entity-wrapper-access-single-entity-reference-2.patch
+  - cd sites/all/modules/
```
Comment (amitaibu) [inline .travis.yml:37]: can we avoid this change

**Comment by RoySegall** [comment]
@amitaibu Until the patch for Entity API will be merge in travis will need to clone the project and apply the patch.

File: .travis.yml (hunk 1 above)
Comment (e0ipso) [inline .travis.yml:37]: I'm not sure of the benefit of cloning the dev branch against `drush dl --dev --yes entity`

Comment (RoySegall) [inline .travis.yml:37]: When using `drush dl --dev --yes entity` i can't apply the patch with `patch -p1`

Comment (e0ipso) [inline .travis.yml:37]: Hmmm, I wonder why that is.

Comment (RoySegall) [inline .travis.yml:37]: Something with this error: https://travis-ci.org/RESTful-Drupal/restful/builds/51050586#L238

Comment (RoySegall) [inline .travis.yml:37]: I try to re role the patch and i get the same error:

```
Hunk #1 FAILED at 821.
```

Comment (amitaibu) [inline .travis.yml:37]: @RoySegall There was a new entity API release, would you mind re-rolling this

**Comment by amitaibu** [comment]
Closed in favor of #428

**Pull request closed without merging**